Encryption

nikitashumsky opened the pull request:
implementation of encryption and decryption methods for byte[]

Nikita-Shumsky committed:
implementation of encryption and decryption methods for byte[]

nikitakoliadin commented:
We MUST merge into DEV branch not MASTER. I have changed it.

nikitakoliadin commented on application/pom.xml:
```diff
@@ -54,6 +54,12 @@
+        <dependency>
+            <groupId>org.bouncycastle</groupId>
+            <artifactId>bcpkix-jdk15on</artifactId>
+            <version>1.58</version>
```
Move the version in mvn properties.

nikitakoliadin commented on application/src/main/java/com/knubisoft/application/ecryption/impl/EncryptionImpl.java:
```diff
@@ -0,0 +1,54 @@
+    @Override
+    @SneakyThrows
+    public byte[] encryptData(byte[] data, X509Certificate encryptionCertificate) {
+        if (nonNull(data) && nonNull(encryptionCertificate)) {
```
If data or encryptionCertificate are nulls - throw exception pls. The same for decrypt method

nikitakoliadin commented on application/src/main/java/com/knubisoft/application/ecryption/Encryption.java:
```diff
@@ -0,0 +1,10 @@
+
+public interface Encryption {
+
+    byte[] encryptData(byte[] rawData, X509Certificate certificate);
```
Can we update encrypt and decrypt method signatures to accept byte[] instead of X509Certificate and PrivateKey?

Nikita-Shumsky committed:
moved bouncycastle version to pom file properties;
methods encryptData() and decryptData now throws exception if one of the arguments is null;

both encryptData() and decrypt data now accepts two byte arrays instead of X509Certificate and PrivateKey

nikitakoliadin reviewed (commented):
Looks good. Good job. We should wait for Vadim with his Unit tests. I want to see it in action.

Commit:
Add annotation @Service to EncryptionImpl
Write test in ApplicationTests for encryption and decryption

Commit:
Fix test and clean up code

Commit:
Clean code

Commit:
Clean code

nikitakoliadin committed:
Merge pull request #4 from nikitakoliadin/testEncryption

Test encryption
